src/hal/i_components/biquad.icomp

```diff
@@ -122,14 +122,14 @@ Biquad_CalcCoeffs(void *arg, unsigned long period)
 {
 struct inst_data *ip = arg;
 
-    double                      sampleRate, w0, alpha;
-    double                      b0, b1, b2, a0, a1, a2;
+    hal_float_t sampleRate, w0, alpha;
+    hal_float_t b0, b1, b2, a0, a1, a2;
 
     // If not direct coefficient loading.
     if(type_ != TYPE_DIRECT){
         valid = 0;
 
-        sampleRate = 1.0 / (period * 1e-9);
+        sampleRate = (hal_float_t) (1.0 / (period * 1e-9));
```

Comment:
better `sampleRate = (1.0 / ((hal_float_t)period * 1e-9));` prevents implicit casting of the long value